src/Compilers/CSharp/Portable/Binder/Binder_Deconstruct.cs

```diff
@@ -695,9 +699,9 @@ private static string ExtractDeconstructResultElementName(BoundExpression expres
         private BoundBadExpression MissingDeconstruct(BoundExpression receiver, SyntaxNode rightSyntax, int numParameters, DiagnosticBag diagnostics,
                                     out ImmutableArray<BoundDeconstructValuePlaceholder> outPlaceholders, BoundExpression childNode)
         {
-            if (!receiver.Type.IsErrorType())
+            if (!(receiver.Type?.IsErrorType() ?? false))
             {
-                Error(diagnostics, ErrorCode.ERR_MissingDeconstruct, rightSyntax, receiver.Type, numParameters);
+                Error(diagnostics, ErrorCode.ERR_MissingDeconstruct, rightSyntax, receiver.Type!, numParameters);
```

Comment:
this feels odd, we explicitly do enter this block when 'receiver.Type' is null. Don't we?

Reply:
I misplaced a paren, thanks.